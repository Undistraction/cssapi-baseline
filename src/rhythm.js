import { CONFIG } from './const';
import { defaultConfig } from './utils';
import { throwError, invalidConfigMessage } from './errors';
import validateConfig from './validators/validateConfig';

const build = config => `xx`;

const configure = (config = defaultConfig(CONFIG)) =>
  validateConfig(config).matchWith({
    Success: build,
    Failure: ({ value }) => {
      throwError(invalidConfigMessage(value));
    },
  });

export default {
  configure,
};
