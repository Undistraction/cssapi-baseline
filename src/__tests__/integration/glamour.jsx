import React from 'react';
import * as glamor from 'glamor';
import renderer from 'react-test-renderer';
import serializer from 'jest-glamor-react';
import rhythm from '../../index';

expect.addSnapshotSerializer(serializer);

const baseline = rhythm.configure();

function Element() {
  const className = glamor.css({
    ...baseline(16),
  });
  return <div className={`${className}`} />;
}

describe(`works with galmour`, () => {
  it(`renders component with correct 'fontSize' and 'lineHeight'`, () => {
    const tree = renderer.create(<Element />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
