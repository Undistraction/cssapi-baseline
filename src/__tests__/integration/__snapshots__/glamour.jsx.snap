// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with galmour renders component with correct 'fontSize' and 'lineHeight' 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 1rem;
  line-height: 1.25rem;
}

<div
  className="glamor-0"
/>
`;
